packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,499 @@
+    private renderHiddenSelectElement(): React.ReactNode {
+        return (
+            <select
+                required={this.props.required || null}
+                name={this.props.name || null}
+                form={this.props.form || null}
+                value={this.state.value}
+                multiple={this.props.multiselectable || null}
+                disabled={this.props.disabled || null}
+                onChange={this.onSelectValueChange}
+                style={{
+                    display: "none",
+                }}
+            />
+        );
+    }
+
+    /**
+     * This exists only to suppress a build warning
+     */
+    private onSelectValueChange = (event: React.ChangeEvent): void => {
+        return null;
+    };
+
+    /**
+     * Determine which function to use to render content display (ie. the part of the control that shows when the menu isn't open)
+     * and invokes it
+     */
+    private renderContentDisplay(): React.ReactNode {
+        if (this.props.trigger !== undefined) {
+            return this.props.trigger(this.props, this.state);
+        } else {
+            return this.defaultDisplayRenderFunction(this.props, this.state);
+        }
+    }
+
+    /**
+     * Determine which function to use to render the menu and invokes it
+     */
+    private renderMenu(): React.ReactNode {
+        if (!this.state.isMenuOpen) {
+            return;
+        }
+        let shouldfocus: boolean = this.props.autoFocus;
+        if (shouldfocus === undefined) {
+            shouldfocus = this.props.multiselectable ? false : true;
+        }
+        return (
+            <Listbox
+                labelledBy={this.props.labelledBy}
+                disabled={this.props.disabled}
+                focusItemOnMount={shouldfocus}
+                multiselectable={this.props.multiselectable}
+                defaultSelection={this.state.selectedItems}
+                selectedItems={this.props.selectedItems}
+                onSelectedItemsChanged={this.updateSelection}
+                managedClasses={{
+                    listbox: get(this.props.managedClasses, "select_menu", ""),
+                    listbox__disabled: get(
+                        this.props.managedClasses,
+                        "select_menuDisabled",
+                        ""
+                    ),
+                }}
+            >
+                {this.props.children}
+            </Listbox>
+        );
+    }
+
+    /**
+     * Updates selection state and associated values
+     */
+    private updateSelection = (newSelection: ListboxItemProps[]): void => {
+        const newValue: string | string[] = this.getValueFromSelection(newSelection);
+        const newDisplayString: string = this.getFormattedDisplayString(newSelection);
+        this.setState({
+            selectedItems: newSelection,
+            value: newValue,
+            displayString: newDisplayString,
+        });
+        if (this.props.onValueChange) {
```

Comment:
I would prefer if this checked for an actual type, eg. `typeof this.props.onValueChange === "function"`

Reply:
yep